sdk/monitor/azquery/custom_client.go

```diff
@@ -109,47 +111,82 @@ func (e *ErrorInfo) Error() string {
 // Row of data in a table, types of data used by service specified in LogsColumnType
 type Row []any
 
-// Table - Contains the columns and rows for one table in a query response.
-type Table struct {
-	// REQUIRED; The list of columns in this table.
-	Columns []*Column `json:"columns,omitempty"`
+// TimeInterval specifies the time range over which to query.
+// Use NewTimeInterval() for help formatting.
+// Follows the ISO8601 time interval standard with most common format being startISOTime/endISOTime.
+// Use UTC for all times.
+type TimeInterval string
```

Comment:
How important is having access to the underlying string?  i.e. are there cases where customers would need to write their own, properly formatted string?  If not, then I suggest we change the type as follows.
```go
type TimeInterval struct {
    Start time.Time
    End time.Time
}
```
I think this is more idiomatic than using a constructor.  It does remove the ability to specify a string though.  So if that's important then leave it as is.

Reply:
Yes, it's very important for the user to have access to the underlying string so they can express durations if necessary. For example, the string "PT2H" means "last two hours" and is a valid value for the TimeInterval.

Reply:
I see.  I think the docs should be updated then to indicate that this can also accept such format as this was not clear to me.
```go
// Also accepts an interval expressed as an ISO8601 duration (e.g. P3Y6M4DT12H30M5S)
```

Reply:
fixed in commit 
f54dc06